src/lib/server/memberProviders/d4hMembersProvider.ts

```diff
@@ -78,6 +78,27 @@ export default class D4HMembersProvider implements MemberProvider {
     }
   }
 
+  async findMember(memberId: string) {
+    await this.initialize();
+    for (const token in this.tokenFetchInfo) {
```

Comment:
Right now, every unit is using the same D4H token, so this might only be N=1, but it's possible for each unit to have their own token, where we'd be making N=(number of units) network requests, hoping that one of the tokens had permissions to complete the request. I think this method should look more like the top of the `getMemberPhoto` method - we should already have the member data cached in the tokenFetchInfo object. `const member = this.tokenFetchInfo[token].lookup[memberId].response`

Reply:
refactored